src/bots/discord/commands/praise.ts

```diff
@@ -0,0 +1,199 @@
+      );
+    }
+    if (receiverData.roleMentions) {
+      embed.addField(
+        'Roles Mentioned',
+        (receiverData.roleMentions?.join(', ') || '') +
+          "\nYou can't dish praise to entire roles."
+      );
+    }
+    embed.addField('Reason', reason || 'No reason entered.');
+    return embed;
+  };
+
+  if (
+    !receivers ||
+    receivers.length == 0 ||
+    !receiverData.validReceiverIds ||
+    receiverData.validReceiverIds?.length == 0
+  ) {
+    const noReceiverEmbed = praiseErrorEmbed(
+      'Receivers not mentiond',
+      'This command requires atleast one valid receiver to be mentioned.'
+    );
+
+    await interaction.reply({ embeds: [addInfoFields(noReceiverEmbed)] });
+    return;
+  }
+
+  if (!reason || reason.length == 0) {
+    const noReasonEmbed = praiseErrorEmbed(
+      'Reason not provided',
+      'Praise needs a `reason` in order to be dished.'
+    );
+    await interaction.reply({ embeds: [addInfoFields(noReasonEmbed)] });
+    return;
+  }
+
+  const User = await UserModel.findOne({
+    accounts: userAccount,
+  });
+
+  if (!User) {
+    const notActivatedEmbed = praiseErrorEmbed(
+      'Account Not Activated',
+      'Your Account is not activated in the praise system. Unactivated accounts can not praise users. Use the `/praise-activate` command to activate your praise account and to link your eth address.'
+    );
+    await interaction.reply({ embeds: [addInfoFields(notActivatedEmbed)] });
+    return;
+  }
+
+  const praised: string[] = [];
+  const receiverIds = receiverData.validReceiverIds.map((id) =>
+    id.substr(3, id.length - 4)
+  );
+  const Receivers = (await guild.members.fetch({ user: receiverIds })).map(
+    (u) => u
+  );
+
+  const guildChannel = await guild.channels.fetch(
+    interaction?.channel?.id || ''
+  );
+
+  for (const receiver of Receivers) {
+    const ra = {
+      id: receiver.user.id,
+      username: receiver.user.username + '#' + receiver.user.discriminator,
+      profileImageUrl: receiver.avatar,
+      platform: 'DISCORD',
+      activateToken: randomstring.generate(),
+    };
+    const receiverAccount = await UserAccountModel.findOneAndUpdate(
+      { username: ra.username },
+      ra,
+      { upsert: true, new: true }
+    );
+
+    const receiverUser = await UserModel.findOne({
+      accounts: receiverAccount,
+    });
+    if (!receiverUser) {
+      await receiver.send(
+        "You were just praised in the TEC! It looks like you haven't activated your account... To activate use the `/praise-activate` command in the server."
+      );
+    }
+
+    await PraiseModel.create({
+      reason: reason,
+      giver: userAccount!._id,
+      sourceId: `DISCORD:${guild.id}:${interaction.channelId}`,
+      sourceName: `DISCORD:${encodeURI(guild.name)}:${encodeURI(
+        guildChannel?.name || ''
+      )}`,
+      receiver: receiverAccount!._id,
+      createdAt: interaction.createdAt,
```

Comment:
The `created_at` is auto created when document is created in mongo. Let's use that automatic date so we don't create confusion if server timezone is different from discord timezone (it most likely is).